Compilation error under chrono-tz feature

**Describe the bug**
<!--
A clear and concise description of what the bug is.
-->

When enabling `chrono-tz` feature, there are compilation error:

```
error[E0061]: this function takes 0 arguments but 2 arguments were supplied
    --> arrow/src/compute/kernels/cast.rs:5646:21
     |
5646 |         let mut b = PrimitiveDictionaryBuilder::new(keys_builder, values_builder);     
     |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ------------  -------------- argument of type `arrow_array::builder::PrimitiveBuilder<arrow_array::types::Int32Type>` unexpected
     |                                                     |                                                                                                                                                                           
     |                                                     argument of type `arrow_array::builder::PrimitiveBuilder<K>` unexpected
     |
note: associated function defined here                                                                             
    --> /Users/liangchi/repos/arrow-rs/arrow-array/src/builder/primitive_dictionary_builder.rs:108:12
     |
108  |     pub fn new() -> Self {                                                                                  
     |            ^^^   
help: remove the extra arguments
     |                                                                                                             
5646 |         let mut b = PrimitiveDictionaryBuilder::new();
     |                                                    ~~                                                
                                                                                                                   
error[E0061]: this function takes 0 arguments but 2 arguments were supplied
    --> arrow/src/compute/kernels/cast.rs:5658:21                                                                  
     |                                                                                                             
5658 |         let mut b = StringDictionaryBuilder::new(keys_builder, values_builder);
     |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ------------  -------------- argument of type `arrow_array::builder::GenericStringBuilder<i32>` unexpected
     |                                                  |                                                          
     |                                                  argument of type `arrow_array::builder::PrimitiveBuilder<K>` unexpected
     |  
```

**To Reproduce**
<!--
Steps to reproduce the behavior:
-->

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

**Additional context**
<!--
Add any other context about the problem here.
-->

--- a/.github/workflows/arrow.yml
+++ b/.github/workflows/arrow.yml
@@ -108,9 +108,12 @@ jobs:
       - name: Check compilation --no-default-features --all-targets --features test_utils
         run: |
           cargo check -p arrow --no-default-features --all-targets --features test_utils
-      - name: Check compilation --no-default-features --all-targets --features --ffi
+      - name: Check compilation --no-default-features --all-targets --features ffi
         run: |
           cargo check -p arrow --no-default-features --all-targets --features ffi
+      - name: Check compilation --no-default-features --all-targets --features chrono-tz
+        run: |
+          cargo check -p arrow --no-default-features --all-targets --features chrono-tz
 
   # test the --features "simd" of the arrow crate. This requires nightly Rust.
   linux-test-simd:

--- a/arrow/src/compute/kernels/cast.rs
+++ b/arrow/src/compute/kernels/cast.rs
@@ -5640,10 +5640,9 @@ mod tests {
     /// Creates a dictionary with primitive dictionary values, and keys of type K
     #[cfg(feature = "chrono-tz")]
     fn make_dictionary_primitive<K: ArrowDictionaryKeyType>() -> ArrayRef {
-        let keys_builder = PrimitiveBuilder::<K>::new();
         // Pick Int32 arbitrarily for dictionary values
-        let values_builder = PrimitiveBuilder::<Int32Type>::new();
-        let mut b = PrimitiveDictionaryBuilder::new(keys_builder, values_builder);
+        let mut b: PrimitiveDictionaryBuilder<K, Int32Type> =
+            PrimitiveDictionaryBuilder::new();
         b.append(1).unwrap();
         b.append(2).unwrap();
         Arc::new(b.finish())
@@ -5652,10 +5651,8 @@ mod tests {
     /// Creates a dictionary with utf8 values, and keys of type K
     #[cfg(feature = "chrono-tz")]
     fn make_dictionary_utf8<K: ArrowDictionaryKeyType>() -> ArrayRef {
-        let keys_builder = PrimitiveBuilder::<K>::new();
         // Pick Int32 arbitrarily for dictionary values
-        let values_builder = StringBuilder::new();
-        let mut b = StringDictionaryBuilder::new(keys_builder, values_builder);
+        let mut b: StringDictionaryBuilder<K> = StringDictionaryBuilder::new();
         b.append("foo").unwrap();
         b.append("bar").unwrap();
         Arc::new(b.finish())

